scripts/packages/VSCodeServer/src/misc.jl

```diff
@@ -139,3 +139,24 @@ function activate_uri(conn, uri)
     Pkg.activate(uri)
     return nothing
 end
+
+function activate_from_dir(conn, dir)
+    path = search_up_file("Project.toml", dir)
+    if path === nothing
+        @warn "No Project.toml found for `$(dir)`"
```

Comment:
I think I'd prefer to show this in the VSCode UI. Maybe return a `(error = true, dir = dir)` NT here instead and show a notification in VSCode?

Reply:
addressed